projects/services-ogc/src/lib/owc/owc-json.service.ts

```diff
@@ -331,195 +460,256 @@ export class OwcJsonService {
+        if (layers.length) {
+          /** if filterType is Baselayers -> create a merged Layer */
+          if (filterType === Filtertypes.Baselayers) {
+            const mergedDescription = layers.map(i => i.description).filter(d => d); // filter empty elements
+            const legendImages = layers.map(i => i.legendImg).filter(d => d);
+            const layerOptions: ILayerOptions = {
+              type: 'custom',
+              id: `${groupName}_${layers.map(i => i.id).join(' ')}`.replace(/\s/g, '_'),
+              name: groupName,
+              mergedLayers: layers,
+              filtertype: Filtertypes.Baselayers
+            };
+            if (mergedDescription.length) {
+              layerOptions.description = mergedDescription.join(';\r\n');
+            }
+            if (legendImages) {
+              layerOptions.legendImg = legendImages[0];
+            }
+
+            const mergedLayer = new Layer(layerOptions);
+            return mergedLayer;
+          } else {
+            const layerGroup = new LayerGroup({
+              id: `${groupName}_${layers.map(i => i.id).join(' ')}`.replace(/\s/g, '_'),
+              name: groupName,
+              layers,
+              filtertype: layers[0].filtertype  // @TODO: can some layers have a different filter-type? -> All layers in a Group must be from the same filter type
+            });
+            return layerGroup;
+          }
+        }
+      }))
+      // making sure no undefined layers are returned
+      .pipe(filter(lg => lg instanceof LayerGroup || lg instanceof Layer));
 
     return layerGroup$;
   }
 
   createLayerFromDefaultOffering(resource: IOwsResource, owc: IOwsContext, targetProjection: string): Observable<Layer> {
-    const offerings = resource.properties.offerings;
-    const offering = offerings.find(o => isWmsOffering(o.code))
-      || offerings.find(o => isWmtsOffering(o.code))
-      || offerings.find(o => isWfsOffering(o.code))
-      || offerings.find(o => isTMSOffering(o.code))
-      || offerings[0];
-    return this.createLayerFromOffering(offering, resource, owc, targetProjection);
+    const offerings = resource.properties?.offerings;
+    if (offerings) {
+      // TODO: allow Multiple offerings ???
+      const offering = offerings.find(o => isWmsOffering(o.code))
+        || offerings.find(o => isWmtsOffering(o.code))
+        || offerings.find(o => isWfsOffering(o.code))
+        || offerings.find(o => isTMSOffering(o.code))
+        || offerings[0];
+      return this.createLayerFromOffering(offering, resource, owc, targetProjection);
+    } else {
+      return of<null>(null);
+    }
   }
 
   createLayerFromOffering(offering: IOwsOffering, resource: IOwsResource, context: IOwsContext, targetProjection: string): Observable<Layer> {
     const layerType = this.getLayertypeFromOfferingCode(offering);
-    if (isRasterLayertype(layerType)) {
+    if (isRasterLayertype(layerType) && isVectorLayertype(layerType)) {
```

Comment:
Here we double check for the type, because TMS and Custom types can be both raster and vector.